### PR TITLE
Fix various "abuses" of filters/properties/attributes

### PR DESCRIPTION
**Pull request by fingolfin**
There is also some generic code in several which uses `ResetFilterObj` on a bunch of properties and filters, e.g. in `homalgResetFilters`; this is also highly questionable and a potential source of bugs, but I see no way (for me, at least) to change that, as it digs deep into the homalg innards.

### CODE REVIEW

File: homalg/gap/HomalgFunctor.gd [hunk 1]
```diff
@@ -156,10 +156,7 @@ DeclareAttribute( "OperationOfFunctor",
 ##  <#/GAPDoc>
 ##
 DeclareAttribute( "Genesis",
-        IsHomalgFunctor );
-
-DeclareAttribute( "IsIdentityOnObjects",
```
Comment (mohamed-barakat) [inline homalg/gap/HomalgFunctor.gd]: These three lines were probably deleted by accident, right?

Comment (fingolfin) [reply homalg/gap/HomalgFunctor.gd]: @mohamed-barakat No, this is intentional. See https://github.com/homalg-project/homalg_project/pull/286/commits/6f1c79fd09732a98faaf1ee7cc9e469c13d30935 (this PR has several commits, each has additional explanations.

**Comment by fingolfin** [comment]
@mohamed-barakat @zickgraf anything I should do with this PR? Do I need to rebase it to let your CI tests run on it or something like that?

**Comment by zickgraf** [comment]
Regarding the CI: it has already been triggered automatically and is configured to run on a "virtual" merge commit, so there is not need to rebase.

**Comment by mohamed-barakat** [comment]
> @mohamed-barakat @zickgraf anything I should do with this PR? Do I need to rebase it to let your CI tests run on it or something like that?

No, I was just waiting for jenkins which is now back again thanks to @zickgraf. I will review the PR before the end of the week.

**Comment by mohamed-barakat** [comment]
Thank you very much Max for the careful work.

**Pull request merged**